packages/mux-uploader/src/mux-uploader-status.ts

```diff
@@ -0,0 +1,77 @@
+import { globalThis, document } from 'shared-polyfills';
+import { getMuxUploaderEl } from './utils/element-utils';
+import { MuxUploaderElementEventMap, IMuxUploaderElement } from './mux-uploader';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+
+:host([upload-error]) .status-message {
+  color: #e22c3e;
+}
+</style>
+
+<span class="status-message" id="status-message" role="status" aria-live="polite"></span>
+`;
+
+class MuxUploaderStatusElement extends globalThis.HTMLElement {
+  statusMessage: HTMLElement | null | undefined;
+  #uploaderEl: IMuxUploaderElement | null | undefined;
+
+  constructor() {
+    super();
+    const shadowRoot = this.attachShadow({ mode: 'open' });
+    shadowRoot.appendChild(template.content.cloneNode(true));
+
+    this.statusMessage = this.shadowRoot?.getElementById('status-message');
+  }
+
+  connectedCallback() {
+    this.#uploaderEl = getMuxUploaderEl(this);
+
+    if (this.#uploaderEl) {
+      this.#uploaderEl.addEventListener('reset', this.clearStatusMessage.bind(this));
+      this.#uploaderEl.addEventListener('uploaderror', this.onUploadError.bind(this));
+      this.#uploaderEl.addEventListener('success', this.onSuccess.bind(this));
+      this.#uploaderEl.addEventListener('uploadstart', this.clearStatusMessage.bind(this));
+    }
+  }
+
+  disconnectedCallback() {
+    if (this.#uploaderEl) {
+      this.#uploaderEl.removeEventListener('reset', this.clearStatusMessage.bind(this));
+      this.#uploaderEl.removeEventListener('uploaderror', this.onUploadError.bind(this));
+      this.#uploaderEl.removeEventListener('success', this.onSuccess.bind(this));
+      this.#uploaderEl.removeEventListener('uploadstart', this.clearStatusMessage.bind(this));
```

Comment:
(blocking): `.bind()` returns a new function each time so this will not work. the listeners will not be removed.

I recommend using `clearStatusMessage = () => {}` and then there is no need for binding...

Reply:
see https://github.com/muxinc/media-chrome/blob/5c7acfbdd0998f6b18e8dc2c01006abfa8ce7b58/src/js/media-time-range.js#L429-L429